testing/src/main/java/com/linecorp/decaton/testing/processor/SerialProcessing.java

```diff
@@ -38,14 +39,14 @@ public void onProduce(ProducedRecord record) {
 
     @Override
     public synchronized void onProcess(TaskMetadata metadata, ProcessedRecord record) {
-        records.computeIfAbsent(record.key(),
+        records.computeIfAbsent(ByteBuffer.wrap(record.key()),
                                 key -> new ArrayList<>()).add(record);
     }
 
     @Override
     public void doAssert() {
         // Checks there's no overlap between two consecutive records' processing time
-        for (Entry<String, List<ProcessedRecord>> entry : records.entrySet()) {
+        for (Entry<ByteBuffer, List<ProcessedRecord>> entry : records.entrySet()) {
```

Comment:
Then we should convert key to string when generating assert message to get readable result here: https://github.com/line/decaton/pull/157/files#diff-141d497c7aafcef2217814fbe1d2ea75d00a4089212776327360030a935d2278L56